src/mapgen.cpp

```diff
@@ -451,15 +465,26 @@ static void GENERATOR_riot_damage( map &md, const tripoint_abs_omt &p )
             int behavior_roll = rng( 1, 100 );
 
             if( behavior_roll <= 20 ) {
-                md.add_field( current_tile, field_fd_blood );
+                if( tile_can_have_blood( md, current_tile, md.has_flag_ter( ter_furn_flag::TFLAG_WALL,
+                                         current_tile ), days_since_cataclysm ) ) {
+                    md.add_field( current_tile, field_fd_blood );
+                }
             } else if( behavior_roll <= 60 ) {
-                place_blood_streaks( md, current_tile );
+                place_blood_streaks( md, current_tile, days_since_cataclysm );
             } else {
-                place_bool_pools( md, current_tile );
+                place_bool_pools( md, current_tile, days_since_cataclysm );
             }
         }
-        if( x_in_y( 1, 2000 ) ) {
-            md.add_field( current_tile, field_fd_fire );
+
+        // Decrease the chance of fire spawning as the Cataclysm progresses to the min fire chance of 1 in 10,000
+        if( x_in_y( 1,  std::min( 2000 + 714 * days_since_cataclysm, 10000 ) ) ) {
+            if( md.has_flag_ter_or_furn( ter_furn_flag::TFLAG_FLAMMABLE, current_tile ) ||
```

Comment:
the "is flammable" check combined with the reduced natural chance might reduce the fire chance *too* much.

As in, there are only like maybe 10% tiles that are particularly flammable on any given map, so this check alone cuts the fire chance 10x. No need to have an extra scaling down to 1/10,000.. i think...

And, to be clear, I don't actively *mind* this change, i'm just not sure this matches your intent.